internal/controller/postgrescluster/pgbackrest_test.go

```diff
@@ -2317,6 +2317,198 @@ func TestReconcilePostgresClusterDataSource(t *testing.T) {
+	cmd := []string{"cmd", "blah"}
+	volumeMounts := []corev1.VolumeMount{{
+		Name: "mount",
+	}}
+	volumes := []corev1.Volume{{
+		Name: "volume",
+	}}
+	dataSource := &v1beta1.PostgresClusterDataSource{
+		// ClusterName/Namespace, Repo, and Options are tested in
+		// TestReconcilePostgresClusterDataSource
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceStorage: resource.MustParse("1Gi"),
+			},
+		},
+		Affinity: &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
+							Key:      "key",
+							Operator: "Exist",
+						}},
+					}},
+				},
+			},
+		},
+		Tolerations: []corev1.Toleration{{
+			Key:      "key",
+			Operator: "Exist",
+		}},
+	}
+	cluster := &v1beta1.PostgresCluster{
+		Spec: v1beta1.PostgresClusterSpec{
+			Metadata: &v1beta1.Metadata{
+				Labels:      map[string]string{"Global": "test"},
+				Annotations: map[string]string{"Global": "test"},
+			},
+			Backups: v1beta1.Backups{PGBackRest: v1beta1.PGBackRestArchive{
+				Metadata: &v1beta1.Metadata{
+					Labels:      map[string]string{"Backrest": "test"},
+					Annotations: map[string]string{"Backrest": "test"},
+				},
+			}},
+			Image:            "image",
+			ImagePullSecrets: []corev1.LocalObjectReference{{Name: "Secret"}},
+		},
+	}
+
+	for _, openshift := range []bool{true, false} {
+		test := "job"
+		cluster.Spec.OpenShift = initialize.Bool(false)
+		if openshift {
+			cluster.Spec.OpenShift = initialize.Bool(true)
+			test = test + "-openshift"
+		}
+		job := &batchv1.Job{}
+		generateRestoreJobIntent(ctx, cluster, configHash, instanceName,
+			cmd, volumeMounts, volumes, dataSource, job)
+
+		t.Run(test, func(t *testing.T) {
+			t.Run("ObjectMeta", func(t *testing.T) {
+				t.Run("Name", func(t *testing.T) {
+					assert.Equal(t, job.ObjectMeta.Name,
+						naming.PGBackRestRestoreJob(cluster).Name)
+				})
+				t.Run("Namespace", func(t *testing.T) {
+					assert.Equal(t, job.ObjectMeta.Namespace,
+						naming.PGBackRestRestoreJob(cluster).Namespace)
+				})
+				t.Run("Annotations", func(t *testing.T) {
+					// configHash is defined as an annotation on the job
+					annotations := labels.Set(job.GetAnnotations())
+					assert.Assert(t, annotations.Has("Global"))
+					assert.Assert(t, annotations.Has("Backrest"))
+					assert.Equal(t, annotations.Get(naming.PGBackRestConfigHash), configHash)
+				})
+				t.Run("Labels", func(t *testing.T) {
+					// instanceName is defined as a label on the job
+					label := labels.Set(job.GetLabels())
+					assert.Equal(t, label.Get("Global"), "test")
+					assert.Equal(t, label.Get("Backrest"), "test")
+					assert.Equal(t, label.Get(naming.LabelStartupInstance), instanceName)
+				})
+			})
+			t.Run("Spec", func(t *testing.T) {
+				t.Run("Template", func(t *testing.T) {
```

Comment:
I definitely would like to take a look at this change at some point. Using this and defining the input as `yaml` seems like it would make the test better.